packages/core/src/config.ts

```diff
@@ -212,3 +212,13 @@ function mergeAutocompleteShorthands(shorthands: Record<string, string | string[
   }
   , {})
 }
+
+export function definePreset<P extends Preset>(preset: PresetFactory<P>): DefinePreset<P>
+
+export function definePreset<P extends Preset>(preset: P): DefinePreset<P>
+
+export function definePreset<P extends Preset, Args extends any[]>(preset: PresetFactory<P, Args>, ...defaultOptions: Args): DefinePreset<P, Args>
+
+export function definePreset<P extends Preset, Args extends any[]>(preset: DefinePreset<P, Args>, ...defaultOptions: Args) {
+  return functional(preset, ...defaultOptions)
```

Comment:
I don't see we need such proxy. Instead, we could just support evaluating functionally preset in core.